packages/amplify-provider-awscloudformation/lib/amplify-service-manager.js

```diff
@@ -197,7 +197,12 @@ async function deleteEnv(context, envName, awsConfig) {
         appId: amplifyAppId,
         environmentName: envName,
       };
-      await amplifyClient.deleteBackendEnvironment(deleteEnvParams).promise();
+      try {
+        await amplifyClient.deleteBackendEnvironment(deleteEnvParams).promise();
+        await amplifyClient.deleteApp({ appId: amplifyAppId }).promise();
+      } catch (e) {
+        context.print.error(`An error occured deleting: ${e.message}`);
```

Comment:
This is not the right behavior. 
One Amplify Console project (appId) can have multiple backend environments in it, and they don't necessarily all belong to the same frontend project. Blindly deleting the app when deleting a backend env can lead to client un-intentionally deleting all the backend envs within this Amplify Console projects. 
Even in the simple case of One Amplify Console project contain two backend envs, and both belong to the same frontend project, this can lead to the CLI incorrectly printing out an error message when the second env is being deleted because the first env deletion already removed the Amplify Console App. 
Because both `amplify delete` and `amplify env delete` will execute the code. 
The right logic should be after deleting an backend environment, you check to see if the Amplify Console project with the AppId still contains any backend envs, if not, proceed to delete the Amplify Console App like this: 
- if the command is `amplify delete` directly delete the console app
- if the command is `amplify env remove`, prompt the client if he wants to delete the Console App, and delete upon confirmation.

Reply:
You are right I think my previous approach was right, I sort of ended up reusing some code from the service manager